package.json

```diff
@@ -24,18 +24,18 @@
     "express": "^4.18.2",
     "mocha": "^10.2.0",
     "standard": "^17.1.0",
-    "supertest": "^6.3.3"
+    "supertest": "^6.3.3",
+    "redoc": "^2.1.1",
```

Comment:
"redoc" and "swagger-ui-dist" should probably go under [optionalDependencies](https://docs.npmjs.com/cli/v6/configuring-npm/package-json?v=true#optionaldependencies), as they are not development dependencies.

Reply:
IIRC the issue reported around redoc was at install time. I forget now, but I remember doing this on purpose. Happy to have you test it out and tell me otherwise, but my thinking here was it was more like a peer dep before npm actually started installing those. If you want it, you need to also install it in your app.

Reply:
What about [peerDependencies](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#peerdependencies)? I see that npm people flip-flopped on whether or not peer dependencies should be installed by default. As of npm v7 they are installed by default :unamused:. I do not see any other package.json dependency type  where the dependencies are not installed by default, but only suggested to be installed.

Reply:
Yeah so the problem reported was an install hang: https://github.com/wesleytodd/express-openapi/issues/38

So my thinking was that we need to avoid installing that package entirely.

Reply:
I guess I didn't test it fully though, so I guess I would need to publish a prerelease and try to install to see which variant would work.

Reply:
Knex has its peer dependencies in `peerDependenciesMeta`, and marks them as optional https://github.com/knex/knex/blob/master/package.json#L81C3 . This does not install the dependencies and emits no warning. There has to be more info in the readme, so that people relying on the UIs know what to do.

Reply:
Oh yeah, good call. That should for sure be addressed. Are you interested in me adding you on the repo so we can collab?

Reply:
> Knex has its peer dependencies in peerDependenciesMeta

Yeah, I think there are plenty of ways to address this, but I guess honestly I am nearing the idea of just dropping the redoc dep entirely? Thoughts on that?

Reply:
We do not use redoc. I would appreciate its removal.
You can add me on the repo. Thanks!

Reply:
Ok, updated this to remove redoc. Will add you now. I did end up moving it back to a direct dep just since it was easy, but left the logic to make things optional. Thoughts on even just droping this whole PR and instead just having a new one to remove redoc?

Reply:
Agreed, let's take care of the redoc removal first.